### PR TITLE
Feat: Archlinux added as Docker Image + Poetry adjustment + Fix: Commit Train squash 

### PR DESCRIPTION
**Pull request by tiaguinho-code**
# Docker Image
Archlinux (A Linux Distribution like f.e Ubuntu) replaces the basic python Docker image for Development environment and final packaging of project.

# Commit Fix
Squashed my commit spam on main into a single one.

# .zshrc
Added .zshrc for configuration of docker image, will export to other Docker customization focused repo soon.

# Poetry
- Removed torch (for now)
- Added ipykernel (Jupytr)

### CODE REVIEW

File: .zshrc
Comment (mlederbauer) [inline .zshrc:1]: Why again do we need this file? We were discussing that it would be best to keep only the project dependencies in the github reository and no bash-related stuff. Perhaps we could create our docker image with these aliases in place, but not as part of this github repo? Please excuse if I oversaw something!

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
-COPY poetry.toml pyproject.toml ./
-RUN poetry install
+# Install python, poetry and looks stuff from arch system repos
+RUN pacman -Syu python python-poetry ranger neovim eza git tree zsh openssh which neofetch github-cli --noconfirm
 
+# Set working directory and copy over config files and instl
+RUN mkdir -p /home/steve
+WORKDIR /home/steve/NMRcraft
+RUN git clone https://github.com/mlederbauer/NMRcraft.git /home/steve/NMRcraft
+RUN poetry install
```
Comment (mlederbauer) [inline Dockerfile:10]: Super, thank you; I trust you with the logic in this one! Could you tell me why we are running “poetry install” and not “make install” here? As seen in ./Makefile, ‘make install’ also installs pre-commit hooks in addition to packages with poetry.

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+# Quality of Life stuff
+RUN chsh -s $(which zsh)
+RUN sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+RUN cp /home/steve/NMRcraft/.zshrc /root/.zshrc
+VOLUME [ "/home/steve" ]
 
-# Use the python binary of the local environment
-CMD ["bash"]
+# start a zsh shell
+CMD ["zsh"]
```
Comment (mlederbauer) [inline Dockerfile:19]: You are more of an expert in this, could we move the contents of ./.zshrc here? Along the lines of moving the “quality of life stuff” there to the “quality of life stuff” here

File: README.md [hunk 3]
```diff
@@ -41,6 +41,20 @@ WIP #TODO
 
 To develop this project, install the environment and the pre-commit hooks with
 
+## Getting started with your project
```
Comment (mlederbauer) [inline README.md]: Why do we need this change to the README? I understand that the comments that are written here are for creating a new repository (“init commit”). For our case, we already haveour repository, and need to clone it locally (e.g., via the command git clone https://… etc). Why do we need it or could we also revert the changes made in this section?

File: pyproject.toml [hunk 4]
```diff
@@ -18,13 +18,13 @@ matplotlib = "^3.8.3"
 numpy = "^1.26.4"
 scipy = "^1.12.0"
 pandas = "^2.2.1"
-torch = "^2.2.1"
```
Comment (mlederbauer) [inline pyproject.toml:21]: Yes I think we can leave torch out of the project for now, thank you for looking at it!

**Comment by mlederbauer** [comment]
Thank you Tiago, see attached for my comments !!

File: Dockerfile (hunk 2 above)
Comment (tiaguinho-code) [reply Dockerfile:19]: Not really as the stuff would escape, I'm sending it to another repo.

File: README.md (hunk 3 above)
Comment (tiaguinho-code) [reply README.md]: This was an accident and I didn't mean to touch this.

File: pyproject.toml (hunk 4 above)
Comment (tiaguinho-code) [reply pyproject.toml:21]: Yep, we can add it back later

File: .zshrc
Comment (tiaguinho-code) [reply .zshrc:1]: Oh no, this was from before the discussion and more of a quick and dirty fix before any commits on main get pushed as that would make it way harder to get rid of the extra commits (from what I read)

File: Dockerfile (hunk 1 above)
Comment (tiaguinho-code) [reply Dockerfile:10]: I was not familiar yet with that functionality but I can gladly investigate it for future implementation of the pre commit script

Comment (mlederbauer) [reply Dockerfile:10]: No problem at all, it’s no important. I'd suggest that if “make install” works as a command, we could replace this line wiht it. If it causes any issues, let’s leave it as-is

File: Dockerfile (hunk 2 above)
Comment (mlederbauer) [reply Dockerfile:19]: Ok thanks so much! No problem if that’s too much trouble

**Review by mlederbauer — APPROVED** [review]

**Comment by mlederbauer** [comment]
ready to merge from my side ! Kindly let me know if this works for you now :))

File: Dockerfile (hunk 1 above)
Comment (tiaguinho-code) [reply Dockerfile:10]: Ok, I looked into it and just copied the subsection of the make install. Using make install in the docker script doesn't work as it spawns a poetry shell at the end of the install section, which leads to a crash on the docker as it is not a interactive shell while installing. I ported over the three commands that work and will include it in the next feature pull request I do on the docker.

File: README.md (hunk 3 above)
Comment (tiaguinho-code) [reply README.md]: Thanks for fixing this ;)
